src/connectors/range/connectRange.ts

```diff
@@ -110,40 +181,39 @@ export default function connectRange(renderFn, unmountFn = noop) {
       const isValidNewNextMin = isFiniteNumber(newNextMin);
       const isValidMinCurrentRange = isFiniteNumber(currentRangeMin);
       const isGreaterThanCurrentRange =
-        isValidMinCurrentRange && currentRangeMin <= newNextMin;
+        isValidMinCurrentRange && currentRangeMin! <= newNextMin!;
```

Comment:
If this repo tooling supports it, we can start using [`assert`](https://www.typescriptlang.org/docs/handbook/release-notes/typescript-3-7.html#assertion-functions) functions so that the TS compiler understands that for this code path, those variables exist.

Reply:
isFiniteNumber already does that, but since we saved it to variables, it doesn't work in all cases. I didn't want to risk making too many changes here (since the variables were chosen to make the final equation not too complex). Not sure what you prefer

Reply:
Fine—we can introduce this pattern in future PRs. The main advantage is that we don't need to use `!` at every use, but only assert that the variable exists in the scope (it'll also be handy in our `index` widget where the lifecycle mutates the "class variables").

Reply:
changed it to need less !